python/lsst/ap/verify/ingestion.py

```diff
@@ -25,31 +25,343 @@
+        """Ingest callibration images into a calibration repository.
+
+        Parameters
+        ----------
+        repo : `str`
+            The output repository location on disk for raw images. Must exist.
+        calibRepo : `str`
+            The output repository location on disk for calibration files. Must
+            exist.
+        calibDataFiles : `list` of `str`
+            A list of filenames to ingest. Supported files vary by instrument
+            but may include flats, biases, darks, fringes, or sky. May contain
+            wildcards.
+        """
+        # TODO: --output is workaround for DM-11668
+        args = [repo, "--calib", calibRepo, "--output", os.path.join(calibRepo, "dummy"),
+                "--mode", "link", "--validity", "999"]
+        args.extend(calibDataFiles)
+        try:
+            _runIngestTask(self.calibIngester, args)
+        except sqlite3.IntegrityError as detail:
+            raise_from(RuntimeError("Not all calibration files are unique"), detail)
+
+    def _ingestDefects(self, dataset, workspace):
+        """Ingest the defect files for use by LSST.
+
+        After this method returns, the calibration repository in ``workspace``
+        shall contain all defects from ``dataset``. Butler operations on the
+        repository shall not be able to modify ``dataset``.
+
+        Parameters
+        ----------
+        dataset : `lsst.ap.verify.dataset.Dataset`
+            The dataset on which the pipeline will be run.
+        workspace : `lsst.ap.verify.workspace.Workspace`
+            The location containing all ingestion repositories.
+        """
+        if os.path.exists(os.path.join(workspace.calibRepo, "defects")):
+            self.log.info("Defects were previously ingested, skipping...")
+        else:
+            if not os.path.isdir(workspace.calibRepo):
+                os.mkdir(workspace.calibRepo)
+
+            if self.config.defectTarball:
+                self.log.info("Ingesting defects...")
+                defectFile = os.path.join(dataset.defectLocation, self.config.defectTarball)
+                self._doIngestDefects(workspace.dataRepo, workspace.calibRepo, defectFile)
+                self.log.info("Defects are now ingested in {0}".format(workspace.calibRepo))
+            else:
+                self.log.info("No defects to ingest, skipping...")
+
+    def _doIngestDefects(self, repo, calibRepo, defectTarball):
+        """Ingest defect images.
+
+        Parameters
+        ----------
+        repo : `str`
+            The output repository location on disk for raw images. Must exist.
+        calibRepo : `str`
+            The output repository location on disk for calibration files. Must
+            exist.
+        defectTarball : `str`
+            The name of a .tar.gz file that contains all the compressed
+            defect images.
+        """
+        # TODO: clean up implementation after DM-5467 resolved
```

Comment:
Hmm... I glanced briefly at DM-5467 and thought it related to the need to have a tar file and/or a dedicated call to `IngestCalibsTask` instead of just keeping the defects with the other calibs. Since you were involved in that discussion, can you double-check whether or not it's relevant?

Reply:
And if something in the current code *is* DECam-specific, please tell me what.